tests/test_ioc_system.py

```diff
@@ -324,71 +263,91 @@ async def test_create_softioc_time_epics_changes(
-
-    await caput(TEST_PREFIX + ":PCAP:ARM", 1, wait=True, timeout=TIMEOUT)
 
-    # Confirm the server received the expected string
-    assert "*PCAP.ARM=" not in dummy_server_system.expected_message_responses
-
-
-@pytest.mark.asyncio
-async def test_pending_changes_blocks_record_set(
-    dummy_server_system: DummyServer, subprocess_ioc
-):
-    """Test that when a value is Put to PandA and subsequently reported via *CHANGES?
-    does not do another .set() on the record"""
-
-    # Trigger a _RecordUpdater.update(), to do a Put command
-
-    dummy_server_system.expected_message_responses.update(
-        {"PCAP1.TRIG_EDGE=Either": "OK"}
-    )
-
-    await caput(TEST_PREFIX + ":PCAP1:TRIG_EDGE", "Either", wait=True, timeout=TIMEOUT)
-
-    # Confirm the server received the expected string
-    assert not dummy_server_system.expected_message_responses
-
-    dummy_server_system.send += ["!PCAP1.TRIG_EDGE=Either\n.", ".", "."]
+    try:
+        arm_queue = asyncio.Queue()
+        m1 = camonitor(TEST_PREFIX + ":PCAP:ARM", arm_queue.put, datatype=str)
+        assert await asyncio.wait_for(arm_queue.get(), TIMEOUT) == "0"
 
-    async def expected_messages_received():
-        """Wait until the expected messages have all been received by the server"""
-        while len(dummy_server_system.send) > 2:
-            await asyncio.sleep(0.1)
+        await caput(TEST_PREFIX + ":PCAP:ARM", 1, wait=True, timeout=TIMEOUT)
 
-    await asyncio.wait_for(expected_messages_received(), timeout=TIMEOUT)
+        assert await asyncio.wait_for(arm_queue.get(), TIMEOUT) == "1"
+    finally:
+        m1.close()
 
 
 @pytest.mark.asyncio
-async def test_bobfiles_created(dummy_server_system: DummyServer, subprocess_ioc: Path):
-    bobfile_temp_dir = subprocess_ioc
+async def test_bobfiles_created(mocked_panda_standard_responses):
+    bobfile_temp_dir, *_ = mocked_panda_standard_responses
     assert bobfile_temp_dir.exists() and BOBFILE_DIR.exists()
     old_files = os.listdir(BOBFILE_DIR)
     for file in old_files:
-        assert filecmp.cmp(
-            f"{bobfile_temp_dir}/{file}", f"{BOBFILE_DIR}/{file}"
-        ), f"File {bobfile_temp_dir/file} does not match {BOBFILE_DIR/file}"
+        assert (
+            Path(bobfile_temp_dir / file)
+            .read_text()
+            .replace(TEST_PREFIX, "TEST-PREFIX")
+            == (BOBFILE_DIR / file).read_text()
+        )
 
     # And check that the same number of files are created
     new_files = os.listdir(bobfile_temp_dir)
     assert len(old_files) == len(new_files)
+
+
+@pytest.mark.asyncio
+async def test_create_softioc_record_update_send_to_panda(
+    mocked_panda_standard_responses,
+):
+    """Test that updating a record causes the new value to be sent to PandA"""
+    (
+        tmp_path,
+        child_conn,
+        response_handler,
+        command_queue,
+    ) = mocked_panda_standard_responses
+
+    await asyncio.sleep(1.5)
```

Comment:
There's a little inconsistency regarding sleeps at the start of tests. Some tests sleep for 1 second, this one sleeps for 1.5, and others do not sleep at all. It would be good to standardise this - if all tests should be sleeping, then moving it into the fixture would be neater. 

However I'm not immediately sure that this sleep should be necessary at all - the `create_subprocess_ioc_and_responses()` function already explicitly waits for the PandA IOC to have started before signalling that it's ready. So all `caput` should be valid from the get-go.

Reply:
I've now adjusted the subprocesses so that a ready message is sent through a pip when the test is ready to start, the fixture waits for the subprocess to send this. This means we can skip wait messages!